surface/include/pcl/surface/mls.h

```diff
@@ -732,15 +735,13 @@ namespace pcl
       performUpsampling (PointCloudOut &output);
 
     private:
-      /** \brief Boost-based random number generator algorithm. */
-      boost::mt19937 rng_alg_;
+      /** \brief Random number generator algorithm. */
+      mutable std::mt19937 rng_;
 
       /** \brief Random number generator using an uniform distribution of floats
         * \note Used only in the case of RANDOM_UNIFORM_DENSITY upsampling
         */
-      boost::shared_ptr<boost::variate_generator<boost::mt19937&,
-                                                 boost::uniform_real<float> >
-                       > rng_uniform_distribution_;
+      std::unique_ptr<std::uniform_real_distribution<>> rng_uniform_distribution_;
```

Comment:
Any idea why this a ptr? Is it because of the lazy/delayed initialization?

Reply:
It will be only initialized if `upsample_method_` is `RANDOM_UNIFORM_DENSITY`. And value range depends on `search_radius_`, so we have not a fixed value. So I could remove `unique_ptr`, but then we generate a object which we will discard without having used it.